Linkta-org/core
# update eslint and prettier configs; add precommit hook

Co-authored-by: Jordan Woods <ewoods08@gmail.com>


## Description

This PR re-introduces changes initially intended in PR#95 that disappeared into the ether of a rebase merge conflict resolution.  This PR introduces the following:  

1. A pre-commit hook with Husky that will check for TS errors in both the client and server directories and prevent you from committing if there are any errors.  
2. The pre-commit hook also runs lint-staged. a package commonly used with husky pre-commit hooks to run the eslint and prettier files in our ./client and ./server directories .
3. Updates to eslint configs: made changes to better integrate with Prettier and cleaned up config to better reflect our current folder structure ( removed a lot of client configurations from the server eslint config). 
4. Prettier changes:  drastically simplified our Prettier config since most of our declarations were defaults.  

Fixes #238 

## Type of change


## How Can this be tested? Testing Plan to review this PR

1. Checkout this PR and run npm install - new packages are introduced.  
2. You should be able to start the app fromt he root folder or the client and server separately
3.  Go to a any component and make a test change: (e.g. duplicate a component, change a TS value that introduces an error, change our single quotes to double to verify prettier change" and commit.  You should see the pre-commit hook run and commit fail due to TS error. 

Thanks to @ejwoods for all the help in troubleshooting these updates
